packages/heartbeat/src/class-heartbeat.php

```diff
@@ -0,0 +1,244 @@
+		 *    $stats['is-https'] = is_ssl() ? 'https' : 'http';
+		 * });
+		 *
+		 * This will bump the stats for the 'is-https/https' or 'is-https/http' stat.
+		 *
+		 * @param array  $stats The stats to be filtered.
+		 * @param string $prefix The prefix that will automatically be added at the begining at each stat group name.
+		 */
+		$stats  = apply_filters( 'jetpack_heartbeat_stats_array', array(), $prefix );
+		$return = array();
+
+		// Apply prefix to stats.
+		foreach ( $stats as $stat => $value ) {
+			$return[ "$prefix$stat" ] = $value;
+		}
+
+		return $return;
+
+	}
+
+	/**
+	 * Registers jetpack.getHeartbeatData xmlrpc method
+	 *
+	 * @param array $methods The list of methods to be filtered.
+	 * @return array $methods
+	 */
+	public static function jetpack_xmlrpc_methods( $methods ) {
+		$methods['jetpack.getHeartbeatData'] = array( __CLASS__, 'xmlrpc_data_response' );
+		return $methods;
+	}
+
+	/**
+	 * Handles the response for the jetpack.getHeartbeatData xmlrpc method
+	 *
+	 * @param array $params The parameters received in the request.
+	 * @return array $params all the stats that hearbeat handles.
+	 */
+	public static function xmlrpc_data_response( $params = array() ) {
+		// The WordPress XML-RPC server sets a default param of array()
+		// if no argument is passed on the request and the method handlers get this array in $params.
+		// generate_stats_array() needs a string as first argument.
+		$params = empty( $params ) ? '' : $params;
+		return self::generate_stats_array( $params );
+	}
+
+	/**
+	 * Clear scheduled events
+	 *
+	 * @return void
+	 */
+	public function deactivate() {
+		// Deal with the old pre-3.0 weekly one.
+		$timestamp = wp_next_scheduled( 'jetpack_heartbeat' );
+		if ( $timestamp ) {
+			wp_unschedule_event( $timestamp, 'jetpack_heartbeat' );
+		}
+
+		$timestamp = wp_next_scheduled( $this->cron_name );
+		wp_unschedule_event( $timestamp, $this->cron_name );
+	}
+
+	/**
+	 * Interact with the Heartbeat
+	 *
+	 * ## OPTIONS
+	 *
+	 * inspect (default): Gets the list of data that is going to be sent in the heartbeat and the date/time of the last heartbeat
+	 *
+	 * @param array $args Arguments passed via CLI.
+	 *
+	 * @return void
+	 */
+	public function cli_callback( $args ) {
+
+		$allowed_args = array(
+			'inspect',
+		);
+
+		if ( isset( $args[0] ) && ! in_array( $args[0], $allowed_args, true ) ) {
+			/* translators: %s is a command like "prompt" */
+			\WP_CLI::error( sprintf( __( '%s is not a valid command.', 'jetpack' ), $args[0] ) );
+		}
+
+		$stats           = self::generate_stats_array();
+		$formatted_stats = array();
+
+		foreach ( $stats as $stat_name => $bin ) {
+			$formatted_stats[] = array(
+				'Stat name' => $stat_name,
+				'Bin'       => $bin,
+			);
+		}
+
+		\WP_CLI\Utils\format_items( 'table', $formatted_stats, array( 'Stat name', 'Bin' ) );
+
+		$last_heartbeat = \Jetpack_Options::get_option( 'last_heartbeat' );
```

Comment:
Do you think it would make sense to import `Jetpack_Options` and `WP_CLI` and get rid of the excessive `\` throughout the file?